parser.go

```diff
@@ -0,0 +1,100 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+const (
+	insertQuery = "INSERT INTO %s (%s) VALUES (%s)"
+	updateQuery = "UPDATE %s SET %s WHERE %s"
+	deleteQuery = "DELETE FROM %s WHERE %s"
+
+	setStatementSeparator   = ","
+	whereStatementSeparator = "AND"
+)
+
+// Parser parses a record into a CQL query statement and its values
+type Parser struct{}
+
+// BuildInsertQuery takes a record, and returns the insert query statement and values representing that record.
+func (p *Parser) BuildInsertQuery(rec sdk.Record, table string) (string, []interface{}) {
+	keyCols, keyVals, cols, vals := p.getColumnsAndValues(rec.Key.(sdk.StructuredData), rec.Payload.After.(sdk.StructuredData))
+	cols = append(cols, keyCols...)
+	vals = append(vals, keyVals...)
+	query := fmt.Sprintf(insertQuery, table, strings.Join(cols, ", "), p.getPlaceholders(len(cols)))
+	return query, vals
+}
+
+// BuildUpdateQuery takes a record, and returns the update query statement and values representing that record.
+func (p *Parser) BuildUpdateQuery(rec sdk.Record, table string) (string, []interface{}) {
+	keyCols, keyVals, cols, vals := p.getColumnsAndValues(rec.Key.(sdk.StructuredData), rec.Payload.After.(sdk.StructuredData))
+	setStatement := p.pairValuesWithPlaceholder(cols, setStatementSeparator)
+	whereStatement := p.pairValuesWithPlaceholder(keyCols, whereStatementSeparator)
+	vals = append(vals, keyVals...)
+	query := fmt.Sprintf(updateQuery, table, setStatement, whereStatement)
+	return query, vals
+}
+
+// BuildDeleteQuery takes a record, and returns the delete query statement and values representing that record.
+func (p *Parser) BuildDeleteQuery(rec sdk.Record, table string) (string, []interface{}) {
+	keyCols, keyVals, _, _ := p.getColumnsAndValues(rec.Key.(sdk.StructuredData), rec.Payload.After.(sdk.StructuredData))
+	whereStatement := p.pairValuesWithPlaceholder(keyCols, whereStatementSeparator)
+	query := fmt.Sprintf(deleteQuery, table, whereStatement)
+	return query, keyVals
+}
+
+// getPlaceholders returns a string of question marks seperated by a comma with a given length.
+func (p *Parser) getPlaceholders(length int) string {
+	return strings.TrimSuffix(strings.Repeat("?, ", length), ", ")
+}
+
+func (p *Parser) pairValuesWithPlaceholder(cols []string, separator string) string {
+	var result string
+	for i := 0; i < len(cols); i++ {
+		result += cols[i] + " = ? " + separator + " "
+	}
```

Comment:
Nitpick: Building a string in a loop will be more performant using a [`strings.Builder`](https://pkg.go.dev/strings#Builder). That said, in this specific case the code can be simplified with the use of [`strings.Join`](https://pkg.go.dev/strings#Join) (which already uses `strings.Builder` [under the hood](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/strings/strings.go;l=427)):
```
return strings.Join(s, " = ? "+separator+" ") + " = ?"
```